x-pack/metricbeat/module/aws/utils.go

```diff
@@ -85,11 +85,10 @@ func GetListMetricsOutput(namespace string, regionName string, period time.Durat
 			for _, metric := range page.Metrics {
 				metricWithAccountID = append(metricWithAccountID, MetricWithID{metric, monitoringAccountID})
 			}
-			return metricWithAccountID, nil
-		}
-
-		for i, metric := range page.Metrics {
-			metricWithAccountID = append(metricWithAccountID, MetricWithID{metric, page.OwningAccounts[i]})
+		} else {
+			for i, metric := range page.Metrics {
+				metricWithAccountID = append(metricWithAccountID, MetricWithID{metric, page.OwningAccounts[i]})
+			}
```

Comment:
What about replacing the two almost identical loops with something like this?

```go
for paginator.HasMorePages() {
    page, err := paginator.NextPage(context.TODO())
    if err != nil {
        return metricWithAccountID, fmt.Errorf("error ListMetrics with Paginator, skipping region %s: %w", regionName, err)
    }

    for i, metric := range page.Metrics {
        owningAccount := monitoringAccountID
        if page.OwningAccounts != nil {
            owningAccount = page.OwningAccounts[i]
        }
        metricWithAccountID = append(metricWithAccountID, MetricWithID{metric, owningAccount})
    }
}
```

Reply:
oh yes... I got too excited to just remove the return 😂 will make the change!